post_processing/pylbo/visualisation/spectra/spectrum_merged.py

```diff
@@ -38,13 +38,24 @@ def __init__(self, data, figsize, custom_figure, interactive, legend, **kwargs):
         )
         self.data = data
         self.leg_handle = LegendHandler(interactive)
-        super()._set_plot_properties(kwargs)
+        kwargs_plt = dict(kwargs)
+        kwargs_plt.pop("colors", None)
+        kwargs_plt.pop("color_parameter", None)
+        super()._set_plot_properties(kwargs_plt)
         self._use_legend = legend
         self._single_color = False
         if isinstance(kwargs.get("color", None), str):
             self._single_color = True
             # if everything is 1 color no use for a legend
             self._use_legend = False
+        self._color_from_parameter = False
+        self._colors = None
+        self._color_parameter = False
+        if isinstance(kwargs.get("colors", None), dict):
+            if isinstance(kwargs.get("color_parameter", None), str):
+                self._color_from_parameter = True
+                self._colors = kwargs.get("colors", None)
+                self._color_parameter = kwargs.get("color_parameter", None)
```

Comment:
Again a bit confusing, what do these do exactly? I'd suggest handling them through the dedicated `plot_props` attribute of the superclass